### ISSUE TITLE
Setting TextDecoration = None on TextBlock does not remove the strike through after applying it

### ISSUE THREAD

**Issue by knightmeister** [microsoft/microsoft-ui-xaml#1093]
**Describe the bug**
I am trying to strike through a TextBlock when the user taps it. This works great, but it's not possible to clear the strike through decoration once it's been applied.

**Steps to reproduce the bug**
1. Add a TextBlock to a page
2. Add a Tapped event
3. Add the following code:

```
        private void TextBlock_Tapped(object sender, TappedRoutedEventArgs e)
        {
            var tb = (TextBlock)sender;
            if(tb.TextDecorations == Windows.UI.Text.TextDecorations.None)
            {
                tb.TextDecorations = Windows.UI.Text.TextDecorations.Strikethrough;
                tb.Foreground = new SolidColorBrush(Colors.Gray);
            }
            else
            {
                tb.TextDecorations = Windows.UI.Text.TextDecorations.None;
                tb.Foreground = new SolidColorBrush(Colors.Black);
            }
        }
```

**Expected behavior**
Tapping the TextBlock applies a strike through and changes the colour to gray. That works. Tapping it again should change the colour to black and remove the strike through. The colour changes but the strike through isn't removed.

**Version Info**
1903

NuGet package version: 
N/A

**Question**
Is there a work around to this, because it's core for the UI I'm working on.

**Comment by mrlacey** [comment]
@knightmeister Yes, this totally looks like a bug in the `TextBlock` control. As a guess, I'd say it's not looking to do anything when the decoration is set to None, when it should also look to remove anything that's already there.

Fortunately, there's a simple workaround:

```diff
    private void TextBlock_Tapped(object sender, TappedRoutedEventArgs e)
    {
        var tb = (TextBlock)sender;
        if (tb.TextDecorations == TextDecorations.None)
        {
            tb.TextDecorations = Windows.UI.Text.TextDecorations.Strikethrough;
            tb.Foreground = new SolidColorBrush(Colors.Gray);
        }
        else
        {
+           var text = tb.Text;
+           tb.Text = string.Empty;
            tb.TextDecorations = TextDecorations.None;
+           tb.Text = text;
            tb.Foreground = new SolidColorBrush(Colors.Black);
        }
    }
```

**Comment by codendone** [comment]
Yes, this is an old bug.  The TextDecoration property incorrectly does not mark the element as needing rendering when its value changed.

**Comment by knightmeister** [comment]
Thanks for the work around @mrlacey, works well.

**Comment by chingucoding** [comment]
I think this is something that can only be fixed with WinUI 3.

**Comment by tinodo** [comment]
Is there any ETA on a fix?
The workaround is nice, but not if the Text or Content and the TextDecoration are data-bound.
Something like this;
`<TextBlock Text="{x:Bind sys:String.Format(x:Null, '{0} ({1}/{2})', Title, SuccessCount, FailureCount)}" TextDecorations="{x:Bind IsEnabled, Converter={StaticResource StrikethroughConverter}}" />`

A more permanent fix would be great!

**Comment by c-npolyak** [comment]
It has not been fixed in WinUI3. Please fix it - should be a very easy fix and the bug is very nasty

**Comment by duncanmacmichael** [comment]
Fixed in 1.4.0. @bpulliam please close this issue as I cannot :)
